cpp/modules/luma.gl/core/src/model.h

```diff
@@ -32,24 +32,30 @@
 
 namespace lumagl {
 
+/// \brief Collection of keys that provide additional attribute context when present in input scheme metadata.
+// TODO(ilija@unfolded.ai): Document metadata properties and the behavior they trigger once we finalize the list
+struct AttributePropertyKeys {
+ public:
+  /// \brief Determines whether attribute step mode will be instanced or vertex.
+  static const char* IS_INSTANCED;
+};
+
 // TODO(ilija@unfolded.ai): Move out and revisit
 struct UniformDescriptor {
   size_t elementSize;
   bool isDynamic{false};
 };
 
-/// \brief Holds shaders compiled and linked into a pipeline
+/// \brief Holds shaders compiled and linked into a pipeline.
 class Model {
  public:
   class Options;
 
-  /// \brief construct a new Model
-  explicit Model(std::shared_ptr<wgpu::Device> device, const Options&);
-
-  // TODO(ilija@unfolded.ai): Remove once integration is complete and layers provide valid options
-  explicit Model(std::shared_ptr<wgpu::Device> device);
+  /// \brief construct a new Model.
+  Model(wgpu::Device device, const Options&);
 
   void setAttributes(const std::shared_ptr<garrow::Table>& attributes);
+  void setInstancedAttributes(const std::shared_ptr<garrow::Table>& attributes);
   void setUniformBuffers(const std::vector<wgpu::Buffer>& uniforms);
```

Comment:
Should we make setUniformBuffers private or remove it?

Reply:
Sorry I mixed up with the old attributes function

- we certainly do need a way to setUniformBuffers, the problem is of coursing matching our various buffers with the layout indices. For now we only have main layer shader buffer and project module buffer so we can hard code 0 and 1.